core/variant.cpp

```diff
@@ -3065,7 +3065,7 @@ bool Variant::hash_compare(const Variant& p_variant) const {
 				return false;
 
 			for(int i = 0; i < l.size(); ++i) {
-				if(! l[0].hash_compare(r[0]))
+				if(! l[i].hash_compare(r[i]))
```

Comment:
@RebelliousX Line 2956 has the same bug. I can make a PR for this, or you can roll it in this one.